### ISSUE TITLE
Video choppy when streaming on YouTube and Netflix

### ISSUE THREAD

**Issue by Simon-Davies** [QubesOS/qubes-issues#3622]
#### Qubes OS version:

Qubes 4.0 R4.0

#### Affected TemplateVMs: 

fedora-26

---

### Steps to reproduce the behavior:

Enable Intel graphics due to Nvidia GTX 1050 not working with the Noveau or proprietary driver

Install rpmfusion free and nonfree repositories

Install ffmpeg

Stream YouTube or Netflix 

### Expected behavior:

Video isn't choppy 

### Actual behavior:

Video is very choppy 

### General notes:

---

**Comment by andrewdavidwong** [comment]
Has it ever not been choppy on your hardware?

Also, just a shot in the dark, but does setting `audio_low_latency = true` in `/etc/qubes/guid.conf` help at all?

See: https://www.qubes-os.org/doc/config-files/

**Comment by Simon-Davies** [comment]
I'm running Qubes on a new Dell XPS 15 so I doubt it's a hardware issue. I wondered if it was a general Fedora issue.

I'll try that setting and report back. Thanks for your help :)

**Comment by Simon-Davies** [comment]
I tried setting audio_low_latency = true in /etc/qubes/guid.conf but the video is still choppy unfortunately

**Comment by martin-paulus** [comment]
Which X11 graphics driver is being used?

Example on (bare metal) Fedora 27, Dell Latitude E5530:
`grep -i driver /var/log/Xorg.0.log`

```
[..]
[     7.107] (II) intel: Driver for Intel(R) Integrated Graphics Chipsets:
[     7.107] (II) intel: Driver for Intel(R) HD Graphics
[     7.107] (II) intel: Driver for Intel(R) Iris(TM) Graphics
[     7.107] (II) intel: Driver for Intel(R) Iris(TM) Pro Graphics
[     7.109] (II) intel(0): Using Kernel Mode Setting driver: i915, version 1.6.0 20171023
[     7.119] (II) intel(0): [DRI2]   DRI driver: i965
[     7.119] (II) intel(0): [DRI2]   VDPAU driver: va_gl
[..]
```

So, I am using the i915 intel driver. But I am pretty sure that vesa or fbdev drivers would also "work", albeit with lesser performance.

**Comment by feosam** [comment]
When I watch a video and I move with the cursor the video jerkys. If it is a fast Video (for example a car video or a youtube video with 720p) it jerkys also whitout moving the cursor. I didn't have this problem at os 3.2 - even when I was playing videos simultaneously.

I looked at the qubes os homepage for a solution:

https://www.qubes-os.org/doc/intel-igfx-troubleshooting/

The solution would be to add a line (iommu=no-igfx) to the /boot/efi/EFI/qubes/xen.cfg .

This line (iommu=no-igfx) was already in the .cfg file, so I had nothing to add.

I tried also debian 9 - the same problem.

Regards

**Comment by awokd** [comment]
From a suggestion in the mailing list, could you please try "turning off the Intel speed stepping" in your UEFI configuration?

**Comment by feosam** [comment]
There is no configuration for Intel speed step or EIST in the uefi bios.

**Comment by Simon-Davies** [comment]
I turned of Intel Speed Stepping in my Bios but still have the issue.

**Comment by e6lk7dqzm83p** [comment]
I have this issue as well, mainly with VLC.  I've found YouTube videos to generally play well.  With both the higher resolution/larger the video (screen size wise) the worse it is.

Possibly related: fast scrolling causes a glitch in the display.  

All issues much worse on an external display (even if it is the only display turned on and connected via mini-DisplayPort or HDMI to DVI) or on built-in display (I'm using a laptop) at its native (3.2K) resolution.  Dropping the resolution and using the built-in display seems to significantly improve things.

**Comment by e6lk7dqzm83p** [comment]
I noticed that my VLC video issues are worse when I am in full screen mode (as opposed to just maximizing the window); I'm not sure if that's useful information.

**Comment by e6lk7dqzm83p** [comment]
I don't know if this will help at all, but I get the following when I use VLC to play a video:

> VLC media player 3.0.3 Vetinari (revision 3.0.2-225-gc9e3360dd4)
[00005eedf36b2150] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
[00005eedf376a900] vlcpulse audio output error: digital pass-through stream connection failure: Not supported
[00005eedf376a900] main audio output error: module not functional
[000070b080d660a0] main decoder error: failed to create audio output
[00005eedf376a900] vlcpulse audio output error: digital pass-through stream connection failure: Not supported
[00005eedf376a900] main audio output error: module not functional
[000070b080d660a0] main decoder error: failed to create audio output
libEGL warning: DRI2: failed to authenticate
Failed to open VDPAU backend libvdpau_nvidia.so: cannot open shared object file: No such file or directory
libEGL warning: DRI2: failed to authenticate

**Comment by openStove** [comment]
I'm experiencing the same problem on Lenovo T440
00:02.0 VGA compatible controller: Intel Corporation Haswell-ULT Integrated Graphics Controller (rev 0b) (prog-if 00 [VGA controller])
iommu=no-igfx is also already set on Qubes R4.0

**Comment by dill-shower** [comment]
Please reopen this Issue. The problem completely persists in the latest 4.2. Just in case, I tried to pass the GPU to hvm, install the nvidia drivers there, but still a very similar problem remains. The same games, for example, if you specify the resolution ultraHD/4k in the windowed mode, just fly, fps 60-100 and higher, the same glx gears give 3000 fps, which is very good. But the larger we make the window size with this video/animation/whatever else that is updated, the lower the fps drops. When set to fullscreen, the fps drops to very low values, at which use is difficult.

Possibly useful information: disabling vsync in the game settings significantly increases fps. With vsync (2-3 fps), without vsync (20-30 fps). But this is still SIGNIFICANTLY lower than natively on an OS without Qubes OS.

**Comment by dill-shower** [comment]
For experiment I use PVH qube fedora41. YouTube and chrome browser. Fullscreen – freeeeeezeeees, lags. On windows mode with small player (1/8 monitor maybe) – play nice. Tests and top logs

**On fullscreen player, 720p**

```
top - 01:18:00 up  4:52,  2 users,  load average: 6.77, 5.67, 4.89
Tasks: 250 total,   1 running, 249 sleeping,   0 stopped,   0 zombie
%Cpu(s): 45.2 us,  5.4 sy,  0.1 ni, 30.0 id,  0.0 wa,  2.6 hi,  0.2 si, 16.4 st 
MiB Mem :  13705.5 total,   7864.6 free,   3353.1 used,   3005.6 buff/cache     
MiB Swap:   1024.0 total,   1024.0 free,      0.0 used.  10352.5 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                
   1453 user      12  -8   33.2g 517664 171820 S 239.4   3.7      6,36 /usr/lib64/ungoogled-chromium/chromium-browser --type=gpu-process --enable-gpu-rasterization --enable-native-gpu-memory-buffers --use-angle=gl --disable-breakpad --change-stack-guar+ 
   6493 user      20   0 1394.1g 533684 182896 S  69.5   3.8   1:03.78 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   6322 user      20   0 1392.4g 301160 127564 S  19.5   2.1   2:38.76 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   1371 user      12  -8   32.8g 396136 215760 S  19.2   2.8  78:39.40 /usr/lib64/ungoogled-chromium/chromium-browser --enable-native-gpu-memory-buffers --enable-gpu-memory-buffer-video-frames --enable-zero-copy --ignore-gpu-blocklist --disable-gpu-dri+ 
   1020 user      19  -1  330720 134864 124904 S  14.6   1.0  44:21.25 /usr/libexec/Xorg :0 -nolisten tcp vt07 -wr -config xorg-qubes.conf                                                                                                                    
   6604 user      20   0  466436  44988  34772 S   5.6   0.3   0:02.19 /usr/bin/xfce4-terminal                                                                                                                                                                
    820 root      20   0    6292   4184   3928 S   3.3   0.0   1:33.21 /usr/bin/qubes-gui -d 0                                                                                                                                                                
   1652 user      12  -8 1392.4g 159324 113340 S   3.3   1.1  13:13.07 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --extension-process --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-ras+ 
     78 root      20   0       0      0      0 S   2.6   0.0  24:09.67 [xen-balloon]                                                                                                                                                                          
   4809 user      25   5 1392.4g 187252 124668 S   2.0   1.3   2:27.67 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   4896 user      25   5 1392.4g 166196 117688 S   2.0   1.2   1:27.74 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   4848 user      25   5 1392.4g 172024 120204 S   1.7   1.2   1:30.84 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   1455 user      20   0   32.5g 105204  78620 S   1.3   0.7   3:00.94 /usr/lib64/ungoogled-chromium/chromium-browser --type=utility --utility-sub-type=network.mojom.NetworkService --lang=en-US --service-sandbox-type=none --use-angle=gl --use-gl=angle + 
    901 user       9 -11   94300  12432   8040 S   1.0   0.1   0:04.69 /usr/bin/pipewire                                                                                                                                                                      
   1643 user      12  -8 1392.4g 311780 105720 S   0.7   2.2   3:09.20 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --extension-process --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-ras+ 
   1692 user      20   0   32.6g  74792  56752 S   0.7   0.5   0:06.23 /usr/lib64/ungoogled-chromium/chromium-browser --type=utility --utility-sub-type=audio.mojom.AudioService --lang=en-US --service-sandbox-type=none --use-angle=gl --use-gl=angle --ch+ 
   1706 user       9 -11  109100  14068   7620 S   0.7   0.1   0:04.17 /usr/bin/pipewire-pulse                                                                                                                                                                
   4905 user      25   5 1392.4g 172384 121204 S   0.7   1.2   1:32.70 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-raster-threads=3 --enab+ 
   5002 user      12  -8 1394.0g 116108  78184 S   0.7   0.8   0:38.92 /usr/lib64/ungoogled-chromium/chromium-browser --type=renderer --extension-process --change-stack-guard-on-fork=enable --enable-gpu-memory-buffer-video-frames --lang=en-US --num-ras+ 
     19 root      20   0       0      0      0 I   0.3   0.0   0:07.67 [rcu_preempt]                                                                                                                                                                          
    785 root      20   0  889536  59152  57232
```



**On window mode with small player, 720p**


 ```
top - 01:19:09 up  4:53,  2 users,  load average: 8.13, 6.33, 5.18
Tasks: 251 total,   3 running, 248 sleeping,   0 stopped,   0 zombie
%Cpu(s): 33.3 us,  7.1 sy,  0.0 ni, 38.1 id,  0.0 wa,  2.4 hi,  0.0 si, 19.0 st 
MiB Mem :  13646.4 total,   7806.4 free,   3356.2 used,   3028.8 buff/cache     
MiB Swap:   1024.0 total,   1024.0 free,      0.0 used.  10290.2 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                
   1453 user      12  -8   33.2g 511416 173952 S 154.5   3.7      6,38 chromium-browse                                                                                                                                                                        
   6503 user      20   0 1392.5g 458508 183376 S 100.0   3.3   0:40.30 chromium-browse                                                                                                                                                                        
   6322 user      20   0 1392.4g 340760 127492 R  54.5   2.4   2:56.75 chromium-browse                                                                                                                                                                        
   1371 user      12  -8   32.9g 396408 216060 S  18.2   2.8  79:08.42 chromium-browse                                                                                                                                                                        
   1020 user      19  -1  330720 134928 124968 R   9.1   1.0  44:31.83 Xorg                                                                                                                                                                                   
   1652 user      12  -8 1392.4g 159580 113212 S   9.1   1.1  13:17.73 chromium-browse                                                                                                                                                                        
      1 root      20   0   23472  13940   9728 S   0.0   0.1   0:02.05 systemd                                                                                                                                                                                
      2 root      20   0       0      0      0 S   0.0   0.0   0:00.01 kthreadd                                                                                                                                                                               
      3 root      20   0       0      0      0 S   0.0   0.0   0:00.00 pool_workqueue_release                                                                                                                                                                 
      4 root       0 -20       0      0      0 I   0.0   0.0   0:00.00 kworker/R-kvfree_rcu_reclaim      

```

**Comment by jamke** [comment]
I confirm, their is a major problem with youtube video playback on R4.2.
- Fedora-39-based template:
I noticed, that if I run firefox with youtube with template based on fedora-39-minimal, then fullscreen video playback of youtube is flawless and SMOOTH. 
In a window mode it lags. No frame drops, but lags noticeably.
- Fedora-42-based template:
If I change template from f39 to the same but based on fedora-42-minimal, than even fullscreen video playback LAGS very noticeably. No frame drops, but lags, like 10 fps instead of 60. Uncomfortable to watch for sure (major problem for me). In small window it does not lag. In usual size window it lags.

So, TL;DR: There is a huge playback regression somewhere: changing template from fedora-39-* to fedora-42-* in R4.2 bring playback LAGS.

**Comment by jamke** [comment]
@dill-shower can you please  try video playback in fullscreen in qube with template  fedora-39? And fedora-40.
It's not supported by now, but for test purposes it is fine to try.
If lags go away, that would show the same problem that I have.

**Comment by jamke** [comment]
Update: 
I tried `fedora-39` (official template) and `youtube` playback in `firefox` (v119) in full-screen was smooth and flawless.
When template is changed to `fedora-40-xfce` (also official template) playback in `firefox` (v132) in the same qube, the same firefox user profile in full-screen the **playback is LOW fps**, like 10 frames per second or something instead of 30-60.

@andrewdavidwong, I probably should create a separate issue with this specific regression in fedora-40 template, right?

**Comment by jamke** [comment]
**Update with workaround**:

I installed flatpak version of Firefox and checked that YouTube **playback on fedora-42 in flatpak version of Firefox is smooth** (like it was on fedora-39).

In the same qube when I open the same video on the same hardware but in native fedora firefox it is much worse (low-fps playback). 

I think, Qubes OS devs should address the issue, because it probably affects all users, who plays YouTube or other media in Firefox with default installation of Qubes OS.

**Comment by andrewdavidwong** [comment]
> [@andrewdavidwong](https://github.com/andrewdavidwong), I probably should create a separate issue with this specific regression in fedora-40 template, right?

Sure, please feel free to do so. (Especially with the additional information in your latest comment, it looks like it might be distinct enough from this issue to merit a separate one.)

**Comment by DemiMarie** [comment]
I suspect the problem is codecs.  Fedora must use OpenH264 for patent reasons.  Mozilla and Flathub are under no such constraint.

**Comment by jamke** [comment]
I also suspect codecs/libs. But I do not know how to force firefox in fedora-42 template to use the same good codecs as flatpak version does. I mean I installed RPMFusion and media sections from it, including ffmpeg instead of ffmpeg-free, but it did not improve playback in native firefox.

99% of Qubes OS users are probably affected by this issue, and think that lags (low frame rate) of playback is a problem of Qubes OS  or weak CPU, but in fact it isn't.

**Comment by jamke** [comment]
Reported as a separate issue here: https://github.com/QubesOS/qubes-issues/issues/10363
